### PR TITLE
Update file format page

### PR DESCRIPTION
**Pull request by panquez**

### CODE REVIEW

File: docs/try/installation.md [hunk 1]
```diff
@@ -1,159 +1,47 @@
...
-
-### Compiling RINGMesh
-
-RINGMesh need several third parties that are automatically compiled and installed by compiling 
-the project `SUPERBUILD.sln`.
-
- * Open the project `SUPERBUILD.sln` with the visual studio version that you choose during the configuration step.
- * Build the solution.
+ 1. cloning the directory from GitHub
+ 1. downloading Source archives (Released or lining)
```
Comment (panquez) [inline docs/try/installation.md]: what do you mean by "lining"?

File: docs/try/installation.md [hunk 2]
```diff
@@ -1,159 +1,47 @@
...
+please extract the files in the folder where you want to use it.
 
-## Mac OS
 
-### Configuring RINGMesh
+## Develop with RINGMesh
 
-#### Using clang (without Xcode)
-As in Linux.
+RINGMesh is a classical library that provide an API with functions that can be used in an external project.
```
Comment (panquez) [inline docs/try/installation.md]: I am not a big fan of "**classical** library"

File: docs/try/installation.md (hunk 1 above)
Comment (francoisbonneau) [reply docs/try/installation.md]: I means living but I am not sure about the terminology. we have released versions and a living one that aglomerate all new devs.

File: docs/features/file_formats.md [hunk 3]
```diff
@@ -1 +1,69 @@
-# Under construction
+# Input and Output file formats
+
+The GeoModel is the Data Model powered by RINGMesh (see the full description of the GeoModel [here](/features/geomodel).
+RINGMesh proposes an home-made file format with the `.gm` extension to read and write GeoModels.
+
+Beyond that, RINGMesh is able to deal with various file format. We present here an exhaustive list of input and output file format
```
Comment (benjaminchauvin) [inline docs/features/file_formats.md]: "file format**s**"

Comment (benjaminchauvin) [reply docs/features/file_formats.md]: twice

Comment (benjaminchauvin) [reply docs/features/file_formats.md]: actually three times

File: docs/features/file_formats.md [hunk 4]
```diff
@@ -1 +1,69 @@
...
+
+## Input 
+
+### GeoModel2D 
+
+| Extension          |     Software    | 
+| -------------      | -------------   |
+| .gm                | RINGMesh        |
+| .model             | Stradivarius    |
+| .svg               | Inkscape        |
```
Comment (benjaminchauvin) [inline docs/features/file_formats.md]: I do not think that Inkscape owns the SVG file format. Maybe you can write "Vector Graphics Editor (e.g., Inkscape, Adobe Illustrator, etc.)"?

File: docs/features/file_formats.md [hunk 5]
```diff
@@ -1 +1,69 @@
...
+| .msh (in progress) | GMSH            |
+
+### GeoModel3D
+
+| Extension          |     Software    |
+| -------------      | -------------   |
+| .gm                | RINGMesh        |
+| .ml                | SKUA-GOCAD      |
+| .so                | SKUA-GOCAD      |
+| .msh (in progress) | GMSH            |
```
Comment (benjaminchauvin) [inline docs/features/file_formats.md:19]: MSH format will be done soon? If no maybe we should not mention formats that are not fully functional.

File: docs/features/file_formats.md [hunk 6]
```diff
@@ -1 +1,69 @@
...
+
+
```
Comment (benjaminchauvin) [inline docs/features/file_formats.md:21]: General comment: maybe for each software add a link toward their website.

File: docs/features/file_formats.md [hunk 7]
```diff
@@ -1 +1,69 @@
...
+| .msh               | GMSH            |
+| .tetgen            | TetGen          |
+| .csmp              | CSMP++          |
+| .vtk               | ParView         |
+| .mfem              | MFEM            |
+| .inp               | Abaqus          |
+| .adeli             | Adeli           |
+| .fem               | Feflow          |
+| .smesh             | Smesh           |
+| .stl               |                 |
```
Comment (benjaminchauvin) [inline docs/features/file_formats.md:38]: stl has no software. It makes sense and it is as for Inkscape.

File: docs/features/file_formats.md [hunk 8]
```diff
@@ -1 +1,69 @@
-# Under construction
+# Input and Output file formats
+
```
Comment (benjaminchauvin) [inline docs/features/file_formats.md:2]: General comment : maybe we can show how "easy" it is to add a new file format (to suggest people to participate).

File: docs/try/installation.md [hunk 9]
```diff
@@ -1,159 +1,71 @@
...
-
-### Compiling RINGMesh
-
-RINGMesh need several third parties that are automatically compiled and installed by compiling 
-the project `SUPERBUILD.sln`.
-
- * Open the project `SUPERBUILD.sln` with the visual studio version that you choose during the configuration step.
- * Build the solution.
+ 1. cloning the directory from [GitHub](https://github.com/ringmesh/RINGMesh)
+ 1. downloading an archive of the sources ([Releases(https://github.com/ringmesh/RINGMesh/releases)
```
Comment (benjaminchauvin) [inline docs/try/installation.md:10]: I presume that it will automatically become 2. in the html output.

File: docs/try/installation.md [hunk 10]
```diff
@@ -1,159 +1,71 @@
...
+Once your project is started, you can follow some [tutorials](/try/tutorials) to learn more about the RINGMesh API.
 
-#### Using clang (without Xcode)
-As in Linux.
+## RINGMesh Utilities 
 
-#### Using Xcode IDE
-As in Windows but with the Xcode generator
-(use "-G Xcode" if you use cmake in command lines).
+To access the utilities RINGMesh need to be configured and compiled with the corresponding option.
```
Comment (benjaminchauvin) [inline docs/try/installation.md]: needs

File: docs/try/installation.md [hunk 11]
```diff
@@ -1,159 +1,71 @@
...
+If you did not check that option during the configuration using CMake please go back to the 
+[configure and compile tutorial](/try/tutorials/configure_compile_ringmesh) and select the `RINGMESH_WITH_UTILITIES` option.
+Pre-compiled packages have been built with the `RINGMESH_WITH_UTILITIES` option. 
 
-### Compiling RINGMesh
-You need to install the Mac OS "Command Line Developer Tools".
+After building RINGMesh with utilities a set of executable are generated and can be execute from the RINGMesh install directory:
```
Comment (benjaminchauvin) [inline docs/try/installation.md]: is generated [...] executed

File: docs/try/installation.md [hunk 12]
```diff
@@ -1,159 +1,71 @@
...
-Note: you need gcc/g++ version higher or equal to 4.2 to compile RINGMesh.
-In Mac OS, clang is used.
+ * Window: `RINGMesh\build\ringmesh\bin\CONFIG\**.exe`
+ * Linux: `RINGMesh\build\ringmesh\CONFIG\bin\**.exe`
+  
+Here is the list of the RINGMesh utilities:
 
-#### Using clang (without Xcode)
-As in Linux except for the packages.
+ * ringmesh-convert: load a GeoModel, check its validity and export GeoModel.
```
Comment (benjaminchauvin) [inline docs/try/installation.md]: Sometimes you start with an uppercase, sometimes with a lowercase (for the word "load").

File: docs/try/installation.md [hunk 13]
```diff
@@ -1,159 +1,71 @@
...
-In Mac OS, clang is used.
+ * Window: `RINGMesh\build\ringmesh\bin\CONFIG\**.exe`
+ * Linux: `RINGMesh\build\ringmesh\CONFIG\bin\**.exe`
+  
+Here is the list of the RINGMesh utilities:
 
-#### Using clang (without Xcode)
-As in Linux except for the packages.
+ * ringmesh-convert: load a GeoModel, check its validity and export GeoModel. 
+ Supported input and output file format are listed [here](/features/file_formats).
```
Comment (benjaminchauvin) [inline docs/try/installation.md]: formats

File: docs/try/installation.md [hunk 14]
```diff
@@ -1,159 +1,71 @@
...
+ * Window: `RINGMesh\build\ringmesh\bin\CONFIG\**.exe`
+ * Linux: `RINGMesh\build\ringmesh\CONFIG\bin\**.exe`
+  
+Here is the list of the RINGMesh utilities:
 
-#### Using clang (without Xcode)
-As in Linux except for the packages.
+ * ringmesh-convert: load a GeoModel, check its validity and export GeoModel. 
+ Supported input and output file format are listed [here](/features/file_formats). 
+ Validity checks performed are explained [here](/features/validity).
```
Comment (benjaminchauvin) [inline docs/try/installation.md]: Put "performed" at the beginning of the sentence?

File: docs/try/installation.md [hunk 15]
```diff
@@ -1,159 +1,71 @@
...
+ * ringmesh-edit-infos: Edit information of a GeoModel (such as its name) and save this change in the same file (overwrote) or another one.
```
Comment (benjaminchauvin) [inline docs/try/installation.md]: overwrite?

File: docs/try/installation.md [hunk 16]
```diff
@@ -1,159 +1,71 @@
...
+ * ringmesh-mesh-quality: load a GeoModel and compute its cell quality.
```
Comment (benjaminchauvin) [inline docs/try/installation.md]: You can precise that it works only on tetrahedron meshes.

File: docs/try/installation.md [hunk 17]
```diff
@@ -1,159 +1,71 @@
...
+Here is the list of the RINGMesh utilities:
 
-#### Using clang (without Xcode)
-As in Linux except for the packages.
+ * ringmesh-convert: load a GeoModel, check its validity and export GeoModel. 
+ Supported input and output file format are listed [here](/features/file_formats). 
+ Validity checks performed are explained [here](/features/validity).
+ * ringmesh-edit-infos: Edit information of a GeoModel (such as its name) and save this change in the same file (overwrote) or another one.
+ * ringmesh-mesh-quality: load a GeoModel and compute its cell quality. 
+ Several metrics is available and the results is stored as an Attribute on each cell.
```
Comment (benjaminchauvin) [inline docs/try/installation.md]: are available ... are stored

File: docs/try/tutorials.md [hunk 18]
```diff
@@ -1,7 +1,7 @@
 
 #Tutorials
 ##Basics
- * [Install and Compile](): link to a webinar to compile and install RINGmesh.
+ * [Configure and Compile](tutorials/configure_compile_ringmesh): How to configure and compile the RINGMesh source code.
```
Comment (benjaminchauvin) [inline docs/try/tutorials.md]: I feel that we should not put "the" before "RINGMesh source code" but maybe we should ask to someone else.

File: docs/try/tutorials/configure_compile_ringmesh.md [hunk 19]
```diff
@@ -0,0 +1,181 @@
+# Configure and Compile the RINGMesh source code
```
Comment (benjaminchauvin) [inline docs/try/tutorials/configure_compile_ringmesh.md]: Same comment.

File: docs/try/tutorials/configure_compile_ringmesh.md [hunk 20]
```diff
@@ -0,0 +1,181 @@
...
+
+## RINGMesh Configuration options
+RINGMesh comes with a CMake configuration file. Several options can be selected in order to build the project that feet your needs:
```
Comment (benjaminchauvin) [inline docs/try/tutorials/configure_compile_ringmesh.md]: fits?

File: docs/try/tutorials/configure_compile_ringmesh.md [hunk 21]
```diff
@@ -0,0 +1,181 @@
...
+ * `BUILD_RINGMESH_VIEW`: Build the viewer to visualize the GeoModel data structure.
+ * `RINGMESH_WITH_GRAPHICS`: Build graphical libraries of RINGMesh and GEOGRAM.
+ * `RINGMESH_WITH_TESTS`: Build the RINGMesh test environment.
+ * `RINGMESH_WITH_TETGEN`: Build the [TetGen](http://wias-berlin.de/software/index.jsp?id=TetGen&lang=1) wrapper library. 
+ This option provides functionalities to mesh and re-mesh volume and surfaces with TetGen. Please make sure that you respect the 
+ [AGPLv3 license](http://www.gnu.org/licenses/agpl-3.0.en.html) before checking that option.
+ * `RINGMESH_WITH_TUTORIALS`: Build the tutorial environment.
+ * `RINGMESH_WITH_UTILITIES`: Build all executable corresponding to RINGMesh utilities.
+ 
+## Linux configure and compile
```
Comment (benjaminchauvin) [inline docs/try/tutorials/configure_compile_ringmesh.md]: configuration and compilation?

File: docs/try/tutorials/configure_compile_ringmesh.md [hunk 22]
```diff
@@ -0,0 +1,181 @@
...
+ * CMake (version >= 3.1)
+ * gcc/g++ (version >= 4.8; C++11 support)
+
+#### Optional tools
+ * [Eclipse-cdt](http://www.eclipse.org/cdt/)
+project is provided (.project and .cproject). You can import RINGMesh into
+Eclipse: File>Import...>General>Existing Projects into Workspace.
+ * Doxygen (version >= 1.8.2; C++11 support). This is required if you want to build the documentation.
+ 
+There is no other dependency (everything you need is shipped with RINGMesh).
```
Comment (benjaminchauvin) [inline docs/try/tutorials/configure_compile_ringmesh.md:41]: This is wrong. On Linux several packages have to be installed. We listed them in the past.

File: docs/try/tutorials/configure_compile_ringmesh.md [hunk 23]
```diff
@@ -0,0 +1,181 @@
...
+
+### Configuring RINGMesh
+
+Execute cmake (version >= 3.1) command in a `RINGMesh/build` directory.
```
Comment (benjaminchauvin) [inline docs/try/tutorials/configure_compile_ringmesh.md]: No need to repeat the necessary cmake version. Best to write it just once.

File: docs/try/tutorials/configure_compile_ringmesh.md [hunk 24]
```diff
@@ -0,0 +1,181 @@
...
+
+To compile RINGMesh, go to RINGMesh root directory and:
+
+```bash
+cd build/Release
+make
+```
+To build in debug, go to build/Debug instead.
+
+Note: RINGMesh uses C++ 11 features. You need gcc/g++ version higher or equal to 4.8 to compile it.
```
Comment (benjaminchauvin) [inline docs/try/tutorials/configure_compile_ringmesh.md:71]: Same comment, put that just once in "mandatory tools".

File: docs/try/tutorials/configure_compile_ringmesh.md [hunk 25]
```diff
@@ -0,0 +1,181 @@
...
+
+### Toolbox
+#### Mandatory tools
+ * CMake (version >= 3.1)
+ * gcc/g++ (version >= 4.8; C++11 support)
+ 
+
+#### Optional tools
+
+ * RINGMesh is tested using Visual Studio. Make sure that you have installed C++ package for Visual Studio through the
```
Comment (benjaminchauvin) [inline docs/try/tutorials/configure_compile_ringmesh.md:105]: "using" is weird. "on" "under" instead?

File: docs/try/tutorials/configure_compile_ringmesh.md [hunk 26]
```diff
@@ -0,0 +1,181 @@
...
+
+#### Optional tools
+
+ * RINGMesh is tested using Visual Studio. Make sure that you have installed C++ package for Visual Studio through the 
+Visual Studio installer. It compiles with the following version:
+	* Visual Studio 12 2013 Win64
+	* Visual Studio 14 2015 Win64
+	* Visual Studio 15 2017 Win64
+ * Doxygen (version >= 1.8.2; C++11 support). This is required if you want to build the documentation.
+
```
Comment (benjaminchauvin) [inline docs/try/tutorials/configure_compile_ringmesh.md:111]: Do not repeat the doxygen version which is necessary.

File: docs/try/tutorials/configure_compile_ringmesh.md [hunk 27]
```diff
@@ -0,0 +1,181 @@
...
+There is no other dependency (everything you need is shipped with RINGMesh).
+
+### Configuring RINGMesh
+
+ * Launch CMake GUI interface (version >= 3.1).
```
Comment (benjaminchauvin) [inline docs/try/tutorials/configure_compile_ringmesh.md:116]: The same (to apply everywhere).

File: docs/try/tutorials/configure_compile_ringmesh.md [hunk 28]
```diff
@@ -0,0 +1,181 @@
...
+
+ * Launch CMake GUI interface (version >= 3.1). 
+ * Indicate where is the source code as the path to RINGMesh root and where to put the binaries as 
+ `RINGMesh_root/build`.
+ * Set the Configuration options using the CMake GUI interface.
+ * Launch the `configure`and `generate` option.
+
+### Compiling RINGMesh
+
+RINGMesh need several third parties that are automatically compiled and installed by compiling
```
Comment (benjaminchauvin) [inline docs/try/tutorials/configure_compile_ringmesh.md:124]: needs

File: docs/try/tutorials/configure_compile_ringmesh.md [hunk 29]
```diff
@@ -0,0 +1,181 @@
...
+
+### Compiling RINGMesh
+
+RINGMesh need several third parties that are automatically compiled and installed by compiling 
+the project `SUPERBUILD.sln`.
+
+ * Open the project `SUPERBUILD.sln` with the visual studio version that you choose during the configuration step.
+ * Build the solution.
+ 
+All the RINGMesh third parties have now been compiled, installed and the `RINGMesh.sln` project have been created in the repository: `RINGMesh_root/build/ringmesh`.
```
Comment (benjaminchauvin) [inline docs/try/tutorials/configure_compile_ringmesh.md:130]: In theory RINGMesh is also compiled. Opening RINGMesh.sln is only necessary if you want to develop in RINGMesh and so access the source files directly from VS.

File: docs/features/file_formats.md (hunk 5 above)
Comment (panquez) [reply docs/features/file_formats.md:19]: Yes I am currently working on input and output for both 2D and 3D models. There is a PR on ringmesh repo but it is not perfectly working yet.

**Pull request merged**